File: src/interfaces/canvas_utils_interface.py
# src/interfaces/canvas_utils_interface.py
from abc import ABC, abstractmethod

class CanvasUtilsInterface(ABC):
    @abstractmethod
    def draw_vertical_line(self, context, pos, color='yellow'):
        pass

    @abstractmethod
    def draw_horizontal_line(self, context, pos, color='green'):
        pass

    @abstractmethod
    def draw_center_ruler(self, context, color='blue', line_length=10, spacing=10):
        pass

File: static/js/interfaces/canvasUtilsInterface.js
// static/js/interfaces/canvasUtilsInterface.js
export default class CanvasUtilsInterface {
    drawVerticalLine(_context, _pos, _color) {
        throw new Error('You have to implement the method drawVerticalLine!');
    }
    drawHorizontalLine(_context, _pos, _color) {
        throw new Error('You have to implement the method drawHorizontalLine!');
    }
    drawCenterRuler(_context, _color, _lineLength, _spacing) {
        throw new Error('You have to implement the method drawCenterRuler!');
    }
}